servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -730,7 +732,15 @@ void resetError() {
         @Override
         public Publisher<Collection<E>> discover(final U u) {
             return delegate().discover(u)
-                    .beforeOnError(status::nextError)
+                    .beforeOnError(t -> {
+                        if (LOGGER.isDebugEnabled()) {
+                            LOGGER.warn("Observed an error from {} while discovering '{}':", delegate(), u, t);
+                        } else {
+                            LOGGER.warn("Observed an error from {} while discovering '{}': {}",
+                                    delegate(), u, rootCause(t));
```

Comment:
I considered not logging it if publisher didn't emit any `onNext` events, but decided to not over complicate the logic. If it's a permanent error, like `UnknownHostException`, users have to stop the app, reconfigure it, and run again. If it's a short blip, it won't hurt if we log it here and then fail the request. Without debug level we log only type + msg of the root cause.